lib/pure/options.nim

```diff
@@ -108,6 +108,41 @@ proc get*[T](self: Option[T]): T =
     raise UnpackError(msg : "Can't obtain a value from a `none`")
   self.val
 
+proc get*[T](self: Option[T], otherwise: T): T =
+  ## Returns the contents of this option or `otherwise` if the option is none.
+  if self.isSome:
+    self.val
+  else:
+    otherwise
+
+
+proc each*[T](self: Option[T], callback: proc (input: T)) =
```

Comment:
This is more akin to a `for` loop than a map. A `map` comes with the overhead of allocating a return value, while an `each` does not. It also means consumers don't need to discard the result.